en/docs/deploy/api-microgateway/passing-enduser-attributes-to-the-backend-using-jwt.md

```diff
@@ -1,250 +1,154 @@
-[AbstractMGWClaimRetriever](https://github.com/wso2/product-microgateway/blob/v3.2.0/components/micro-gateway-jwt-generator/src/main/java/org/wso2/micro/gateway/jwt/generator/AbstractMGWClaimRetriever.java) class.
-For that, the user can have the same maven dependency used for custom JWT Generation implementation. A sample 
-implementation is available [here](https://github.com/wso2/product-microgateway/tree/master/samples/sample-jwt-generator).
-In the end, you can add the corresponding jar file to the microgateway project's `/lib` directory prior to executing 
-`micro-gw build` command.
-
-```xml tab="Maven dependency"
-<dependencies>
-    <dependency>
-        <groupId>org.wso2.am.microgw</groupId>
-        <artifactId>mgw-jwt-generator</artifactId>
-        <version>3.2.0</version>
-    </dependency>
-</dependencies>
-```
+    Click here for a sample [Custom Gateway JWTGenerator](https://github.com/wso2/samples-apim/tree/master/CustomGatewayJWTGenerator).
+
+2. Build your class and add the JAR file in the `<MG_HOME>/resources/enforcer/dropins` directory.
+
+3. Enable and configure the JWT implementation.
+
+     - For more information, see [JWT generation configuration details](#jwt-generation-configuration-details).
 
-!!! note
-    Please note that if you have used any additional dependencies to the customized implementation, you may need to add all
-    of them to the microgateway project's `/lib` directory. This is because if the microgateway can't find 
-    those dependencies within the project, the functionality will not be available.
+     - Set `enforcer.jwtGenerator.gatewayGeneratorImpl` to your customized class name.
 
-### Setting the authentication header of the backend request
+        ```toml
+        [enforcer.jwtGenerator]
+        gatewayGeneratorImpl = "org.wso2.carbon.test.CustomGatewayJWTGenerator"
+        ```
 
-When you are passing a token to the backend or when you are generating a JWT inside Microgateway, you can configure the header in which the JWT token will be passed to the backend. The following configuration should be added to the Microgateway runtime configuration in `micro-gw.conf`. (By default the header is set to X-JWT-Assertion)
+4. Start the server.
+
+     For more information, see the [Quick Start Guide]({{base_path}}/deploy/api-microgateway/getting-started/quick-start-guide/quick-start-guide-overview/).
+    
+## JWT generation configuration details
+
+The following is a sample configuration.
 
 ```toml
-[jwtConfig]
-    # JWT header when forwarding the request to the backend
+[enforcer.jwtGenerator]
+    enable = true
+    encoding = "base64" # base64,base64url
+    claimDialect = "http://wso2.org/claims"
     header = "X-JWT-Assertion"
+    signingAlgorithm = "SHA256withRSA"
+    gatewayGeneratorImpl = "org.wso2.carbon.test.CustomGatewayJWTGenerator"
+    publicCertificatePath = "/home/wso2/security/truststore/mg.pem"
+    privateKeyPath = "/home/wso2/security/keystore/mg.key"
```

Comment:
@Rajith90 , shouldn't we have a separate config catalog for the MGW related configurations in the WSO2 API Manager docs?

Reply:
We are planning to add a config catalog section under the microgateway sub topic. Do you think we need to have it under APIM?

Reply:
+1 to that idea and I guess we can have an overview page for all the config catalogs and link to the MGW config catalog from this location https://apim.docs.wso2.com/en/4.0.0/reference/config-catalog/